### DISCUSSION TITLE
cabal: Missing dependency: reflex-dom-pandoc >= 0.7.9.0

### DISCUSSION THREAD

**Discussion by mrzv** [srid/neuron#658] [Installation]
Neuron works great on Linux, but it's incredibly complicated to build Nix on macOS, so the recommended way of building neuron is inaccessible there. I'd like to avoid the docker route, so #528 doesn't help.

When I try to build neuron with `cabal build`, it gives an error that it cannot find `reflex-dom-pandoc >= 0.7.9.0`, and indeed the latest release I see is [0.6.0](https://github.com/srid/reflex-dom-pandoc/releases/tag/0.6.0.0). Where can I find the version that neuron needs?

**Comment by srid** [comment]
@mrzv Last I checked installing Nix on macOS [was straightforward](https://nixos.org/manual/nix/stable/#sect-macos-installation); has that changed?

See https://github.com/srid/neuron/tree/master/dep for all the source-based Haskell dependencies used by neuron. 

Per https://github.com/srid/neuron/blob/master/dep/reflex-dom-pandoc/github.json - you need `63ff62ee9c7f4a8c231db16bb91b858c8a39aebd` of https://github.com/srid/reflex-dom-pandoc

**Reply by srid** [reply]
I suppose it makes sense to release all these libraries to Hackage and also cut a formal 2.0 release of neuron, now that it is in [maintenance mode with new focus on Emanote](https://neuron.zettel.page/next).

**Reply by mrzv** [reply]
That would be fantastic.

**Reply by srid** [reply]
Just pushed reflex-dom-pandoc and pandoc-link-context to Hackage.  @mrzv Let me know if you hit any further roadblock to 'cabal install' way of installing neuron.

**Reply by srid** [reply]
As for neuron 2.0 release I have been hoping to resurrect the old client-side search feature before making it. So it is on hold for now.

**Reply by mrzv** [reply]
All the dependencies built with cabal, so that's a major progress. Thanks!

But now I get the following error, when building neuron itself:
```
...
[25 of 70] Compiling Neuron.Markdown  ( src/Neuron/Markdown.hs, dist/build/Neuron/Markdown.o, dist/build/Neuron/Markdown.dyn_o )

src/Neuron/Markdown.hs:62:7: error:
    • No instance for (Ord Aeson.Value)
        arising from the 'deriving' clause of a data type declaration
      Possible fix:
        use a standalone 'deriving instance' declaration,
          so you can specify the instance context yourself
    • When deriving the instance for (Ord ZettelMeta)
   |
62 |       Ord,
   |       ^^^
cabal: Failed to build neuron-1.9.35.0 (which is required by exe:neuron from
neuron-1.9.35.0). See the build log above for details.
```

**Reply by srid** [reply]
We need a [newer version](https://github.com/haskell/aeson/blob/master/changelog.md#1520) of `aeson`. This should do it: 69cd5bd7

**Reply by mrzv** [reply]
Progress, but another error:
```
...
[42 of 70] Compiling Neuron.CLI.Search ( src/Neuron/CLI/Search.hs, .../neuron/dist-newstyle/build/x86_64-osx/ghc-8.10.6/neuron-1.9.35.1/build/Neuron/CLI/Search.o, .../neuron/dist-newstyle/build/x86_64-osx/ghc-8.10.6/neuron-1.9.35.1/build/Neuron/CLI/Search.dyn_o )

src/Neuron/CLI/Search.hs:20:24: error:
    • staticWhich: Could not find executable for "neuron-search"
    • In the untyped splice: $(staticWhich "neuron-search")
   |
20 | neuronSearchScript = $(staticWhich "neuron-search")
   |                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^
cabal: Failed to build neuron-1.9.35.1 (which is required by exe:neuron from
neuron-1.9.35.1).
```

**Reply by srid** [reply]
This patch is supposed to address that https://github.com/srid/neuron/pull/429/files (it got reverted later on due to a regression)

But for the moment the easiest option is to try building by specifying [that script](https://github.com/srid/neuron/blob/master/neuron-search) in PATH, as in: 

```
PATH=/path/to/neuron/source/tree cabal build
```

**Reply by mrzv** [reply]
Progress, but one more error:
```
Building executable 'neuron' for neuron-1.9.35.1..
[1 of 1] Compiling Main             ( exe/Main.hs, .../neuron/dist-newstyle/build/x86_64-osx/ghc-8.10.6/neuron-1.9.35.1/x/neuron/build/neuron/neuron-tmp/Main.o )

exe/Main.hs:16:3: error:
    Ambiguous occurrence ‘hSetBuffering’
    It could refer to
       either ‘Relude.hSetBuffering’,
              imported from ‘Relude’ at exe/Main.hs:11:1-13
              (and originally defined in ‘Relude.Lifted.Handle’)
           or ‘System.IO.hSetBuffering’,
              imported from ‘System.IO’ at exe/Main.hs:12:19-31
              (and originally defined in ‘GHC.IO.Handle’)
   |
16 |   hSetBuffering stdout LineBuffering
   |   ^^^^^^^^^^^^^

exe/Main.hs:17:3: error:
    Ambiguous occurrence ‘hSetBuffering’
    It could refer to
       either ‘Relude.hSetBuffering’,
              imported from ‘Relude’ at exe/Main.hs:11:1-13
              (and originally defined in ‘Relude.Lifted.Handle’)
           or ‘System.IO.hSetBuffering’,
              imported from ‘System.IO’ at exe/Main.hs:12:19-31
              (and originally defined in ‘GHC.IO.Handle’)
   |
17 |   hSetBuffering stderr LineBuffering
```

**Reply by srid** [reply]
Can you try 62ebb5e3 ?

**Reply by mrzv** [reply]
Victory! It compiles and runs. Many thanks.

**Comment by mrzv** [comment]
The first sentence of the section you link to says that it's not straightforward, specifically that you need a "workaround". The workaround involves creating a new volume, which is less than ideal for many reasons.

Thanks for the pointers. I don't use Haskell and cabal, but hopefully I can figure out how to convince it to use the right commit.

**Reply by srid** [reply]
Right, but I assumed that the command thereafter is supposed to automate all of that. There is another one at https://nix.dev/tutorials/install-nix#macos - and it doesn't mention needing to do any manual workaround prior.

**Reply by mrzv** [reply]
It's a little terrifying to trust a random bit of code downloaded from the Internet to repartition my hard-drive.
